includes/tripal_hq_user_data.form.inc

```diff
@@ -219,20 +219,14 @@ function tripal_hq_entity_form_submit($form, &$form_state) {
   $entity = $controller->entityFormSubmitBuildEntity($form, $form_state);
   $sid = isset($form_state['values']['submission_id']) ? $form_state['values']['submission_id'] : NULL;
 
+
   $values = $form_state['values'];
-  
-  $user_email = $user->mail;
-  $site_email = variable_get('site_mail');
-  $relevant_admins = tripal_hq_get_curator_emails();
 
-  $op = tripal_hq_save_submission($entity, $sid);
-  if ($op === 'inserted') {
+  $sid = tripal_hq_save_submission($entity, $sid);
 
-    //From site, to end user
-    drupal_mail('tripal_hq', 'user_submit_notice', $user_email, language_default(), $values, $site_email, TRUE);
-    //from site to site_admin
-    drupal_mail('tripal_hq', 'admin_submit_notice', $relevant_admins, language_default(), $values, $site_email, TRUE);
+  if ($sid) {
```

Comment:
This no longer makes sense. The else message should indicate an error. Currently, when an error happens it says the record "updated successfully".

Reply:
So you want 2 checks here. The first, to make sure $sid is not false and thus no error happened. The second is
 ```php
if(isset($form_state['values']['submission_id'])) {
  // Tell the user that an update happened
} else {
  // Tell the user that a new submission was created.
}
```